src/thor/trace_route_action.cc

```diff
@@ -167,16 +167,22 @@ thor_worker_t::map_match(Api& request, uint32_t best_paths) {
     const auto& edge_segments = result.segments;
     m_temp_enhanced_match_results.clear();
     m_temp_route_discontinuities.clear();
+    m_temp_disconnected_edges.clear();
+    m_temp_disjoint_edge_groups = {{}};
 
     // Form the path edges based on the matched points and populate disconnected edges
-    m_temp_disconnected_edges.clear();
     m_temp_path_edges = MapMatcher::FormPath(matcher.get(), match_results, edge_segments,
                                              mode_costing, mode, m_temp_disconnected_edges, options);
+    for (const auto& edge : m_temp_path_edges) {
+      m_temp_disjoint_edge_groups.back().emplace_back(edge);
+      if (m_temp_disconnected_edges.count(edge.edgeid) > 0) {
+        m_temp_disjoint_edge_groups.emplace_back();
+      }
+    }
```

Comment:
a description of what this is doing would be great for the sake of clarity. the variable names tell us a little bit but honestly not enough to really grasp what we are doing here. seems we are saying if we see the same edge id twice then we need to start a new group of edges? if that is true then i want to ask what happens when we do a map match that has a loop in the path (which is a completely reasonable thing to occur)?